scopus/author_retrieval.py

```diff
@@ -179,10 +180,15 @@ def subject_areas(self):
         """
         areas = []
         area = namedtuple('Subjectarea', 'area abbreviation code')
-        for item in self._json['subject-areas'].get('subject-area', []):
-            new = area(area=item['$'], code=item['@code'],
-                       abbreviation=item['@abbrev'])
-            areas.append(new)
+        try:
+            for item in self._json.get('subject-areas',
+                                       {}).get('subject-area', []):
+                new = area(area=item['$'], code=item['@code'],
+                           abbreviation=item['@abbrev'])
+                areas.append(new)
+        except AttributeError:
+            # Sometimes subject-areas is mapped to None
+            pass
         return areas
```

Comment:
Can you please add a check that if `areas` is empty it should return `None`?   Some of the surrounding functions have that too.